### PR TITLE
table collection equivalence

### PR DESCRIPTION
**Pull request by mufernando**
WIP #896

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/897/)

**Comment by mufernando** [comment]
I'm currently having (at least) two issues:
- Some valgrind leaks in c/tests/test_tables.c
- I messed up the PyC interface and the python tests are complaining that I'm giving `equivalence` too many arguments.

**Comment by benjeffery** [comment]
Looking good @mufernando! Do you want me to look into the issues you mentioned?

**Comment by mufernando** [comment]
please, Ben!

File: c/tests/test_tables.c [hunk 1]
```diff
@@ -105,6 +105,102 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_provenance_table_add_row(&tc2.provenances, "time", 4, "record", 6);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
```
Comment (petrelharp) [inline c/tests/test_tables.c:201]: you need to free tc1 and tc2 here

File: c/tskit/tables.c [hunk 2]
```diff
@@ -7732,25 +7732,36 @@ tsk_table_collection_free(tsk_table_collection_t *self)
  * tables. We do not consider the file_uuids either, since this is a property of
  * the file that set of tables is stored in. */
 bool
-tsk_table_collection_equals(tsk_table_collection_t *self, tsk_table_collection_t *other)
+tsk_table_collection_equivalence(
+    tsk_table_collection_t *self, tsk_table_collection_t *other, tsk_flags_t options)
```
Comment (petrelharp) [inline c/tskit/tables.c:7736]: I think to match `table_collection_equals` this should be `table_collection_equivalent`?

File: python/tskit/tables.py [hunk 3]
```diff
@@ -2794,6 +2794,28 @@ def union(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
 
+    def equivalence(self, other, compare_metadata=False, compare_provenance=False):
+        """
+        Returns True if  `self` and `other` are equivalent, that is if sequence
+        length and all tables are equal, except for the provenance table.
```
[the remark this answers is not in the document]
Comment (petrelharp) [reply python/tskit/tables.py:2800]: Also mention that equality is if the flags are both True. (Otherwise folks will wonder what the diff is.)

**Review by petrelharp — CHANGES_REQUESTED** [review]

File: c/tests/test_tables.c (hunk 1 above)
Comment (mufernando) [reply c/tests/test_tables.c:201]: yess!!

File: c/tskit/tables.c (hunk 2 above)
Comment (mufernando) [reply c/tskit/tables.c:7736]: agree!

Comment (mufernando) [reply c/tskit/tables.c:7736]: was waiting to do that at the end at once

File: c/tests/test_tables.c [hunk 4]
```diff
@@ -145,60 +145,62 @@ test_table_collection_equivalence(void)
...
-    ret = tsk_table_collection_equivalence(
+    ret = tsk_table_collection_equivalent(
         &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
     CU_ASSERT_FALSE(ret);
     // Copying over tc1 to tc2
     ret = tsk_table_collection_copy(&tc1, &tc2, 0);
     CU_ASSERT_EQUAL_FATAL(ret, 0);
 
-    // Equivalence for different metadata/schema
+    // equivalent for different metadata/schema
```
Comment (petrelharp) [inline c/tests/test_tables.c:155]: do you mean "equivalent *except* for metadata?

File: c/tests/test_tables.c [hunk 5]
```diff
@@ -105,6 +105,102 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    // Equivalence for different metadata/schema
+    ret = tsk_table_collection_set_metadata(
+        &tc1, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
```
Comment (petrelharp) [inline c/tests/test_tables.c:162]: this one should be just `TSK_EQ_CMP_METADATA`, I think? (since the two flags are always used together here, they ought to be separated sometimes)

File: c/tests/test_tables.c [hunk 6]
```diff
@@ -105,6 +105,102 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_set_metadata_schema(
+        &tc1, example_metadata_schema, example_metadata_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
```
Comment (petrelharp) [inline c/tests/test_tables.c:176]: also

File: c/tests/test_tables.c [hunk 7]
```diff
@@ -105,6 +105,102 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
```
Comment (petrelharp) [inline c/tests/test_tables.c]: already did this at L173

File: c/tests/test_tables.c [hunk 8]
```diff
@@ -105,6 +105,102 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
+
+    // Ignore provenance
+    ret = tsk_provenance_table_add_row(&tc1.provenances, "time", 4, "record", 6);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
```
Comment (petrelharp) [inline c/tests/test_tables.c:193]: and these should just use the `TSK_EQ_CMP_PROVENANCE` flag?

File: c/tests/test_tables.c (hunk 4 above)
Comment (mufernando) [reply c/tests/test_tables.c:155]: yes!

File: c/tests/test_tables.c (hunk 5 above)
Comment (mufernando) [reply c/tests/test_tables.c:162]: agree

File: python/tests/test_tables.py [hunk 9]
```diff
@@ -2336,6 +2344,34 @@ def test_copy(self):
...
+        pop_configs = [msprime.PopulationConfiguration(5) for _ in range(2)]
+        migration_matrix = [[0, 1], [1, 0]]
+        t1 = msprime.simulate(
+            population_configurations=pop_configs,
+            migration_matrix=migration_matrix,
+            mutation_rate=1,
+            record_migrations=True,
+            random_seed=1,
+        ).dump_tables()
+        t2 = msprime.simulate(
```
Comment (petrelharp) [inline python/tests/test_tables.py:2357]: why not just copy? this is also testing the ability of msprime to produce the same thing with the same random seed, which is beside the point here.

File: c/tests/test_tables.c (hunk 8 above)
Comment (mufernando) [reply c/tests/test_tables.c:193]: thanks!

File: python/tests/test_tables.py [hunk 10]
```diff
@@ -2336,6 +2344,34 @@ def test_copy(self):
...
+            random_seed=1,
+        ).dump_tables()
+
+        t1.provenances.add_row("random stuff")
+        assert not (t1 == t2)
+        assert t1.equivalent(t2)
+        assert not (t1.equivalent(t2, compare_provenance=True))
+        t1.provenances.clear()
+        t2.provenances.clear()
+        assert t1.equivalent(t2)
```
Comment (petrelharp) [inline python/tests/test_tables.py:2371]: Should also check top-level metadata here, like
```
t1.metadata_schema = tskit.MetadataSchema({"codec": "json", "type": "object"})
t1.metadata = {"hello": "world"}
assert not t1.equivalent(t2)
assert not t2.equivalent(t1)
t2.metadata_schema = t1.metadata_schema
assert not t1.equivalent(t2)
assert not t2.equivalent(t1)
t2.metadata = t1.metadata
assert t1.equivalent(t2)
assert t2.equivalent(t1)
```

File: python/tests/test_tables.py [hunk 11]
```diff
@@ -2336,6 +2344,34 @@ def test_copy(self):
...
+            population_configurations=pop_configs,
+            migration_matrix=migration_matrix,
+            mutation_rate=1,
+            record_migrations=True,
+            random_seed=1,
+        ).dump_tables()
+
+        t1.provenances.add_row("random stuff")
+        assert not (t1 == t2)
+        assert t1.equivalent(t2)
```
Comment (petrelharp) [inline python/tests/test_tables.py:2367]: should check the other way 'round, too:
```
assert t2.equivalent(t1)
```

File: python/tests/test_tables.py (hunk 10 above)
Comment (mufernando) [reply python/tests/test_tables.py:2371]: thank you for this! I was putting off figuring out how to work with the top-level metadata.

Comment (petrelharp) [reply python/tests/test_tables.py:2371]: note: edited that to make it correct just now.

File: c/tests/test_tables.c [hunk 12]
```diff
@@ -105,6 +105,104 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+    ret = tsk_table_collection_equivalent(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_table_collection_equivalent(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_set_metadata_schema(
+        &tc2, example_metadata_schema, example_metadata_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
```
Comment (benjeffery) [inline c/tests/test_tables.c:178]: You can leave out the fatal here and in the other equality tests below and above - it's more informative for when the tests fail to carry on and get output for the other cases you're testing. Generally `FATAL` is used for "something has gone _so_ wrong there's no point continuing"

**Comment by petrelharp** [comment]
Some more suggestions in the tests, but once those are in, this is looking good to go, I think.

**Comment by mufernando** [comment]
It isn't bc if the memory leak and Py-C interface not working heh!

but thanks for all the comments!

File: c/tests/test_tables.c [hunk 13]
```diff
@@ -105,6 +105,104 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_provenance_table_add_row(&tc2.provenances, "time", 4, "record", 6);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalent(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
+    tsk_table_collection_free(&tc1);
```
Comment (benjeffery) [inline c/tests/test_tables.c:196]: Currently the tests only use `TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE` in combination, you need to also test the flags individually.

File: c/tests/test_tables.c [hunk 14]
```diff
@@ -105,6 +105,98 @@ insert_edge_metadata(tsk_table_collection_t *tables)
...
+    ret = tsk_mutation_table_add_row(
+        &tc1.mutations, 0, 0, TSK_NULL, TSK_UNKNOWN_TIME, NULL, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+
+    // Equality of empty vs non-empty
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    // Copying over tc1 to tc2
+    ret = tsk_table_collection_copy(&tc1, &tc2, 0);
```
Comment (petrelharp) [inline c/tests/test_tables.c:152]: here's the other leak: `TSK_NO_INIT`: https://github.com/tskit-dev/tskit/blob/main/c/tskit/tables.h#L2158

**Comment by petrelharp** [comment]
I guess I meant "when it passes tests"...

File: python/_tskitmodule.c [hunk 15]
```diff
@@ -5157,6 +5157,30 @@ TableCollection_has_index(TableCollection *self)
...
 
+static PyObject *
+TableCollection_equivalent(TableCollection *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *ret = NULL;
+    TableCollection *other = NULL;
+    tsk_flags_t options = 0;
+    int cmp_metadata = false;
+    int cmp_provenance = false;
+    static char *kwlist[] = {"compare_metadata", "compare_provenance" , NULL};
```
Comment (petrelharp) [inline python/_tskitmodule.c:5168]: I think you need "other" at the start of kwlist

**Comment by benjeffery** [comment]
I'm reviewing currently, shouldn't be long.

File: python/_tskitmodule.c (hunk 15 above)
Comment (petrelharp) [reply python/_tskitmodule.c:5168]: compare `TableCollection_union`

File: c/tskit/tables.c [hunk 16]
```diff
@@ -7731,26 +7731,42 @@ tsk_table_collection_free(tsk_table_collection_t *self)
...
+bool
+tsk_table_collection_equivalent(
+    tsk_table_collection_t *self, tsk_table_collection_t *other, tsk_flags_t options)
+{
+    bool cmp_metadata = !!(options & TSK_EQ_CMP_METADATA);
+    bool cmp_provenance = !!(options & TSK_EQ_CMP_PROVENANCE);
+    bool ret
+        = self->sequence_length == other->sequence_length
+          && (!cmp_metadata
+                 || (self->metadata_length == other->metadata_length
```
Comment (benjeffery) [inline c/tskit/tables.c:7743]: This part isn't compiling on `gcc 9.3.0` as it is asking for brackets around the second half of the `||`

File: python/_tskitmodule.c (hunk 15 above)
Comment (benjeffery) [reply python/_tskitmodule.c:5168]: Yep, this is needed. Tests for this function pass after adding.

File: python/tests/test_tables.py [hunk 17]
```diff
@@ -112,6 +112,14 @@ def test_low_level_get_row(self):
         with pytest.raises(_tskit.LibraryError):
             t.ll_table.get_row(1)
 
+    def test_low_level_equivalent(self):
```
Comment (benjeffery) [inline python/tests/test_tables.py:115]: This test is looking for an `equivalent` method on each individual table, which I don't think you were planning to implement.

File: python/_tskitmodule.c (hunk 15 above)
Comment (mufernando) [reply python/_tskitmodule.c:5168]: yesssss

**Comment by mufernando** [comment]
@benjeffery should I change union within this PR or create a different one to get more detailed history?

it should be just one line to fix, though maybe I should mess with the metadata in the python tests to make sure we catch this next time?

File: c/tests/test_tables.c (hunk 14 above)
Comment (mufernando) [reply c/tests/test_tables.c:152]: yessssss

**Review by benjeffery — COMMENTED** [review]
Looking good @mufernando, thanks for picking this up!

I'd like us to consider the API of this a bit more. `equivalent` is too close to `equals` for me. If I imagine being a user who is new to tskit then I wouldn't know what it did when reading code that used it. I also don't like how the default behaviour (skipping metadata and provenance) is hidden when someone reads lines like:

```
python: t1.equivalent(t2)
c: tsk_table_collection_equivalent(t1, t2, NULL)
```

Perhaps something like this would make it more explicit what was going on:


```
python: t1.equals(t2, ignore_metadata=True, ignore_provenance=True)
c: tsk_table_collection_equals_with_options(t1, t2, TSK_IGNORE_METADATA | TSK_IGNORE_PROVENANCE)
```

In other words the default is full equality, and you have to be explicit about what you're ignoring. The reader of code then knows exactly what is happening. This also supports future uses cases where the concept of "equivalency" isn't the right concept, for example ignoring mutations. With the flags this way round you don't need a new function in the python c module, just add kwargs to `equals` and add an `equals` method to the `TableCollection`.

**Comment by benjeffery** [comment]
> @benjeffery should I change union within this PR or create a different one to get more detailed history?

Separate PR is cleaner I think. Thanks!

File: c/tskit/tables.c (hunk 16 above)
Comment (mufernando) [reply c/tskit/tables.c:7743]: yes, solved it in a local commit! thansk!

File: c/tests/test_tables.c (hunk 13 above)
Comment (mufernando) [reply c/tests/test_tables.c:196]: yes! changed it

**Comment by petrelharp** [comment]
I think you're right here, @benjeffery. And those method names are pretty good, actually.

File: c/tests/test_tables.c (hunk 12 above)
Comment (mufernando) [reply c/tests/test_tables.c]: thanks for the explanation, Ben! fixed.

**Comment by mufernando** [comment]
I think I'm done!

File: c/tskit/tables.c [hunk 18]
```diff
@@ -7731,26 +7731,41 @@ tsk_table_collection_free(tsk_table_collection_t *self)
...
-               && tsk_node_table_equals(&self->nodes, &other->nodes)
-               && tsk_edge_table_equals(&self->edges, &other->edges)
-               && tsk_migration_table_equals(&self->migrations, &other->migrations)
-               && tsk_site_table_equals(&self->sites, &other->sites)
-               && tsk_mutation_table_equals(&self->mutations, &other->mutations)
-               && tsk_population_table_equals(&self->populations, &other->populations)
-               && tsk_provenance_table_equals(&self->provenances, &other->provenances);
+    bool ret;
+    ret = tsk_table_collection_equals_with_options(self, other, 0);
     return ret;
```
Comment (benjeffery) [inline c/tskit/tables.c:7769]:
```suggestion
    return tsk_table_collection_equals_with_options(self, other, 0);
```

File: python/tskit/tables.py [hunk 19]
```diff
@@ -2794,6 +2794,30 @@ def union(
...
 
+    def equals(self, other, ignore_metadata=False, ignore_provenance=False):
+        """
+        Returns True if  `self` and `other` are equal. The comparison of
+        top-level metadata/metadata schema and provenance tables may be
+        disabled with the flags `ignore_metadata` and `ignore_provenance`,
+        which are true by default.
+
+        :param TableCollection other: Another table collection.
+        :param bool ignore_metadata: If False, the top-level metadata and
```
Comment (benjeffery) [inline python/tskit/tables.py:2805]:
```suggestion
        :param bool ignore_metadata: If True, the top-level metadata and
```

File: python/tskit/tables.py [hunk 20]
```diff
@@ -2794,6 +2794,30 @@ def union(
...
+            metadata schema are ignored.
+        :param bool ignore_provenance: If False, the provenance tables are
```
Comment (benjeffery) [inline python/tskit/tables.py:2807]:
```suggestion
        :param bool ignore_provenance: If True, the provenance tables are
```

**Review by benjeffery — COMMENTED** [review]
Looking good, almost there, just a few quibbles. :+1: 

One more thing (I'm sorry to bring it up, but once things are released names are very hard to change). `TSK_IGNORE_METADATA` sounds like it ignores _all_ metadata including that in the tables within the table collection. I can't think of a better name though. `TSK_IGNORE_TOP_LEVEL_METADATA`?? Too long? Too jargony? I leave it up to you as I'm split on it. Def add an extra note in the docs though, like "Note that table row-level metadata and table schemas are always checked"

**Comment by benjeffery** [comment]
@mufernando Almost forgot! You need changelog entries on both the C and python side.

File: python/tests/test_tables.py [hunk 21]
```diff
@@ -2395,6 +2395,51 @@ def test_equals(self):
...
+        t2.metadata_schema = t1.metadata_schema
+        assert not t1.equals(t2)
+        assert t1.equals(t2, ignore_metadata=True)
+        assert not t2.equals(t1)
+        assert t2.equals(t1, ignore_metadata=True)
+
+        # testing both
+        t1.provenances.add_row("random stuff")
+        assert not t1.equals(t2)
+        assert t1.equals(t2, ignore_metadata=True, ignore_provenance=True)
```
Comment (petrelharp) [inline python/tests/test_tables.py:2436]: Also not equal with just ignoring metadata and then with just ignoring provenance?

**Comment by mufernando** [comment]
Thank you @petrelharp and @benjeffery! Waiting for the tests to pass, but I did everything you all suggested. Still not 100% sure on the `TSK_IGNORE_TOP_LEVEL_METADATA` vs. `TSK_IGNORE_METADATA`. But I am leaning towards more clarity I guess.

**Review by petrelharp — APPROVED** [review]
LGTM!! Thanks, @mufernando!

File: python/tskit/tables.py [hunk 22]
```diff
@@ -2794,6 +2794,31 @@ def union(
...
+    def equals(self, other, ignore_top_level_metadata=False, ignore_provenance=False):
+        """
+        Returns True if  `self` and `other` are equal. The comparison of
+        top-level metadata/metadata schema and provenance tables may be
+        disabled with the flags `ignore_top_level_metadata` and `ignore_provenance`,
+        which are false by default. Note that table row-level metadata and table
+        schemas are always checked.
+
+        :param TableCollection other: Another table collection.
+        :param bool ignore_top_level_metadata: If False, the top-level metadata and
```
Comment (benjeffery) [inline python/tskit/tables.py:2806]: False -> True

File: python/tskit/tables.py [hunk 23]
```diff
@@ -2794,6 +2794,31 @@ def union(
...
+            metadata schema are ignored.
+        :param bool ignore_provenance: If False, the provenance tables are
```
Comment (benjeffery) [inline python/tskit/tables.py:2808]: False -> True

File: c/CHANGELOG.rst [hunk 24]
```diff
@@ -2,6 +2,11 @@
 [0.99.8] - 2020-XX-XX
 ---------------------
 
+**New features**
+
+- Added ``tsk_equals_with_options`` which allows for more flexible equality
```
Comment (benjeffery) [inline c/CHANGELOG.rst:7]: ``tsk_table_collection_equals_with_options``

**Review by benjeffery — APPROVED** [review]
Just one more boolean flip in the docs and then you're good to merge! You can do that yourself by adding the "AUTOMERGE" label. Thanks for pushing this one through @mufernando

**Comment by mufernando** [comment]
Thanks!!

**Comment by benjeffery** [comment]
@mergifyio refresh

**Comment by mufernando** [comment]
@Mergifyio refresh

**Comment by mufernando** [comment]
Had to rebase onto `upstream/main`.

**Comment by benjeffery** [comment]
> Had to rebase onto `upstream/main`.

Thanks, sorry I should have merged this before the big change to the c module.

File: c/tskit/tables.h [hunk 25]
```diff
@@ -718,6 +718,11 @@ typedef struct {
 #define TSK_UNION_NO_CHECK_SHARED (1 << 0)
 #define TSK_UNION_NO_ADD_POP (1 << 1)
 
+/* Flags for table collection equals */
+#define TSK_IGNORE_TOP_LEVEL_METADATA (1 << 0)
+#define TSK_IGNORE_METADATA (1 << 0)
```
Comment (jeromekelleher) [inline c/tskit/tables.h:723]: Just catching up here - does the ``TSK_IGNORE_METADATA`` actually do anything now @mufernando? If not we should file an issue and remove it.

Comment (benjeffery) [reply c/tskit/tables.h:723]: Gah! How'd I miss that. Nice catch!

Comment (benjeffery) [reply c/tskit/tables.h:723]: https://github.com/tskit-dev/tskit/pull/911

Comment (mufernando) [reply c/tskit/tables.h:723]: oops! I think I missed that in the rebase before merging. Thanks for catching it @jeromekelleher and for fixing it @benjeffery !
